distro/run/core/src/test/java/org/camunda/bpm/run/test/plugins/CamundaRunProcessEnginePluginsRegistrationTest.java

```diff
@@ -0,0 +1,126 @@
+import java.util.stream.Collectors;
+
+import org.camunda.bpm.engine.impl.cfg.CompositeProcessEnginePlugin;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.camunda.bpm.engine.impl.cfg.ProcessEnginePlugin;
+import org.camunda.bpm.run.CamundaBpmRun;
+import org.camunda.bpm.run.property.CamundaBpmRunProcessEnginePluginProperty;
+import org.camunda.bpm.run.property.CamundaBpmRunProperties;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = { CamundaBpmRun.class })
+@ActiveProfiles(profiles = { "test-new-plugins" }, inheritProfiles = true)
+public class CamundaRunProcessEnginePluginsRegistrationTest {
+
+  @Autowired
+  protected ProcessEngineConfigurationImpl processEngineConfiguration;
+
+  @Autowired
+  protected CamundaBpmRunProperties properties;
+
+  protected List<ProcessEnginePlugin> plugins;
+
+  @Before
+  public void setUp() {
+    this.plugins = processEngineConfiguration.getProcessEnginePlugins();
+  }
+
+  @Test
+  public void shouldPickUpAllPluginConfigurations() {
+    // given a CamundaBpmRunProperties instance
+    String pluginOne = "org.camunda.bpm.run.test.plugins.TestFirstPlugin";
+    String pluginTwo = "org.camunda.bpm.run.test.plugins.TestSecondPlugin";
+    String pluginThree = "org.camunda.bpm.run.test.plugins.TestDefaultValuesPlugin";
+    List<CamundaBpmRunProcessEnginePluginProperty> pluginConfigs = properties
+        .getProcessEnginePlugins();
+
+    // then
+    // assert that all plugin configuration properties were mapped properly
+    assertThat(pluginConfigs).hasSize(3);
+    List<String> pluginClasses = pluginConfigs.stream()
+        .map(CamundaBpmRunProcessEnginePluginProperty::getPluginClass)
+        .collect(Collectors.toList());
+    assertThat(pluginClasses)
+        .contains(pluginOne, pluginTwo, pluginThree);
+
+    CamundaBpmRunProcessEnginePluginProperty firstPlugin = pluginConfigs.get(0);
+    Map<String, Object> firstPluginParameters = firstPlugin.getPluginParameters();
+    assertThat(firstPluginParameters).hasSize(2);
+    assertThat(firstPluginParameters.keySet()).contains("parameterOne", "parameterTwo");
+    assertThat(firstPluginParameters.values()).contains("valueOne", true);
+
+    CamundaBpmRunProcessEnginePluginProperty secondPlugin = properties.getProcessEnginePlugins().get(1);
+    Map<String, Object> secondPluginParameters = secondPlugin.getPluginParameters();
+    assertThat(secondPluginParameters).hasSize(3);
+    assertThat(secondPluginParameters.keySet())
+        .contains("parameterOne", "parameterTwo", "parameterThree");
+    assertThat(secondPluginParameters.values()).contains(1.222, false, 123);
```

Comment:
```suggestion
    assertThat(secondPluginParameters).containsExactly(
        entry("parameterOne", 1.222),
        entry("parameterTwo", false),
        entry("parameterThree", 123));
```

Reply:
That's cool! I wasn't aware that you could use `contains` with a `Map` in AssertJ. I'm just not sure if the order of the parameters is guaranteed to always be the same. I'll try it out.

Reply:
Oh, you are right, we should use `contains` instead of `containsExactly` as the order is not guaranteed.